code/datums/dna.dm

```
@@ -539,10 +666,18 @@
 	if(!has_dna())
 		return
 	var/num = rand(1, DNA_UNI_IDENTITY_BLOCKS)
-	var/newdna = setblock(dna.uni_identity, num, random_string(DNA_BLOCK_SIZE, GLOB.hex_characters))
-	dna.uni_identity = newdna
+	var/newdna = setblock(dna.unique_identity, num, random_string(DNA_BLOCK_SIZE, GLOB.hex_characters))
+	dna.unique_identity = newdna
 	updateappearance(mutations_overlay_update=1)
 
+/mob/living/carbon/proc/randmutf()
+	if(!has_dna())
+		return
```

Avoid using `nospacescase`, prefer `snake_case` to it.
Also, same as before, make this sanity check `CRASH()`, instead of silently failing.

---

this is not a sanity check - :b:enos don't have dna, but are a subtype of `/mob/living/carbon`

---

But how are they getting a mutation then?
The proc shouldn't be called in the first place.
Let the proc callers deal with that. If they don't it will simply runtime and that will get fixed.

---

I made the sanity check crash like you asked, and during testing, I wasn't able to get it to trigger by way of exposing xenos to unstable mutagen - so yes, your argument is fair.